src/java.desktop/macosx/classes/com/apple/laf/AquaComboBoxUI.java

```diff
@@ -454,6 +454,32 @@ protected LayoutManager createLayoutManager() {
     }
 
     class AquaComboBoxLayoutManager extends BasicComboBoxUI.ComboBoxLayoutManager {
+        protected Rectangle rectangleForCurrentValue() {
+            System.out.println("rectForCurrentValue NEW: " + comboBox.getHeight());
+            int width = comboBox.getWidth();
+            int height = 21;
```

Comment:
Is this a Aqua LAF constraint on max height of text-field?

Reply:
This height limitation is set to match the appearance of a non-oversized font in an editable JComboBox. I tested different values, and this value matches the appearance of an editable JComboBox when the AquaLookAndFeel control font is used.

The reason for overriding this method is to set this max height and condense the logic for the editor rectangle values. Previously, this class hard coded adjustments for the width and height values of the editor rectangle in layoutContainer when editor != null. This occurs right after this rectangleForCurrentValue is called, and this method is only used to calculate the rectangle position. So, overriding this method makes more sense since it does exactly that for Aqua L&F values specifically.